src/Nethermind/Nethermind.Evm/Tracing/CompositeBlockTracer.cs

```diff
@@ -84,11 +83,38 @@ public ITxTracer StartNewTxTrace(Transaction? tx)
 
         public void EndBlockTrace()
         {
-            for (int index = 0; index < _childTracers.Length; index++)
+            for (int index = 0; index < _childTracers.Count; index++)
             {
                 IBlockTracer childTracer = _childTracers[index];
                 childTracer.EndBlockTrace();
             }
         }
+
+        public void Add(IBlockTracer tracer)
+        {
+            _childTracers.Add(tracer);
+            IsTracingRewards |= tracer.IsTracingRewards;
+        }
+        
+        public void AddRange(params IBlockTracer[] tracers)
+        {
+            _childTracers.AddRange(tracers);
+            IsTracingRewards |= tracers.Any(t => t.IsTracingRewards);
+        }
+
+        public void Remove(IBlockTracer tracer)
+        {
+            _childTracers.Remove(tracer);
+            IsTracingRewards = _childTracers.Any(t => t.IsTracingRewards);
+
+        }
+
+        public IBlockTracer GetTracer() =>
```

Comment:
how about passing a type in to GetTracer as an argument and it giving back the tracer which fits that type and null if none of the _childTracer elements have the type?

Reply:
What purpose would it serve? This is intended only for internal BlockchainProcessor use.

Reply:
in that case nevermind